app/models/condition.rb

```diff
@@ -0,0 +1,71 @@
+class Condition < ActiveRecord::Base
+  self.primary_key = 'date'
+  validates :date, :max_temperature_f, :mean_temperature_f, :min_temperature_f, :mean_humidity, :mean_visibility_miles, :max_wind_speed_mph, :precipitation_inches, presence: true
+  has_many  :trips, :foreign_key => 'start_date'
+
+  def self.make_range(attribute, increment)
+    floor = minimum(attribute)
+    ceiling = maximum(attribute)
+    range = []
+
+    until floor >= (ceiling - increment)
+      range << [floor, floor + increment]
+      floor += increment
+    end
+
+    range << [floor, ceiling]
+  end
+
+  def self.chunk(attribute, increment)
+    chunked = {}
+    make_range(attribute, increment).each do |range|
+      chunked[range] = where(attribute => range[0]...range[1])
+    end
+    chunked
+  end
+
+  def self.connect_condition_with_trips(attribute, increment)
+    trips_in_range = {}
+    chunk(attribute, increment).each do |range, conditions|
+      trips_in_range[range] = conditions.joins(:trips).group(:start_date).count.values
+    end
+    trips_in_range
+  end
+
+  def self.ranges_with_trips(attribute, increment)
+    final = {}
+    connect_condition_with_trips(attribute, increment).each do |k, v|
```

Comment:
i suggest choosing more descriptive variable names than `k` and `v`. what do these variables represent?